data/src/stream.rs

```diff
@@ -98,12 +98,19 @@ pub async fn run(server: server::Entry, mut sender: mpsc::Sender<Update>) -> Nev
                         };
                     }
                     Err(e) => {
+                        let e = e.to_string() + &match e {
+                            // unwrap Tls-specific error enums to access more error info
+                            irc::error::Error::Tls(ne) => {
+                                ": ".to_string() + &ne.to_string()
+                            },
+                            _ => "".to_string()
+                        };
```

Comment:
You can do the following which will reduce the # of allocations and is a little easier to read:

```rs
let error = match e {
  irc::Error::Tls(e) => format!("a TLS error occurred: {e}"),
  _ => e.to_string(),
};
```